### ISSUE TITLE
Orderly shutdown of a server, that doesn't interrupt queries

### ISSUE THREAD

**Issue by danielmewes** [rethinkdb/rethinkdb#3978]
Let's say you have a cluster and want to reboot a server.

Even if the server doesn't hold any critical roles (let's say it has only secondary replicas), there currently isn't a way of doing that.
Just shutting down the server interrupts currently running queries.

It would be nice if we could shut down a server such that it stops receiving new queries (any primaries on other servers should stop routing to it), and then waits until all current queries have completed before terminating itself.

This is related to https://github.com/rethinkdb/rethinkdb/issues/2941, except that in this case we're planning to bring the server back later. Ideally the permanent cluster configuration shouldn't be changed by this.

I haven't thought much about how this could interact with auto-failover and Raft in general. If the server is a primary, we might want it to release that role before shutting down.

(@Timer @timmaxw )

**Comment by deontologician** [comment]
What is the expected behavior for open changefeeds?

**Comment by danielmewes** [comment]
Good point. If it's not a primary then there won't be any. But otherwise: Maybe we could interrupt them after some timeout?

**Comment by deontologician** [comment]
could just send `SUCCESS_SEQUENCE` maybe. I don't think the protocol currently explicitly says that's allowed on feeds, but it would probably work.

**Comment by danielmewes** [comment]
Hmm it might be unexpected for a feed to close without an error.

**Comment by danielmewes** [comment]
There's also the question of how to handle new queries on an existing connection directly to that server. Probably we still need to error on those.
Also should we close a client connection as soon as it has no more running queries?
